### PR TITLE
[JW8-5648] Strip object and iframe elements from HTML input

### PR DESCRIPTION
**Pull request by robwalch**
### This PR will...
Address vulnerability to XSS via object and iframe tags in UI elements that accept HTML input.

### Why is this Pull Request needed?
Prevent XSS injection via object tag data and iframes.

#### Addresses Issue(s):
JW8-5648

### CODE REVIEW

**Comment by johnBartos** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  🛠 There are modified src files, but no test changes. Add tests if you're able to.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/js/utils/dom.js [hunk 1]
```diff
@@ -49,10 +49,10 @@ export function htmlToParentElement(html) {
 }
 
 export function sanitizeScriptNodes(element) {
-    const scripts = element.querySelectorAll('script');
-    for (let i = 0; i < scripts.length; i++) {
-        const script = scripts[i];
-        script.parentNode.removeChild(script);
+    const nodes = element.querySelectorAll('script,object,iframe');
```
Comment (jnatalzia) [inline src/js/utils/dom.js]: any tests to update?

File: src/js/utils/dom.js [hunk 2]
```diff
@@ -49,10 +49,10 @@ export function htmlToParentElement(html) {
...
+    for (let i = 0; i < nodes.length; i++) {
+        const node = nodes[i];
+        node.parentNode.removeChild(node);
```
Comment (zetagame) [inline src/js/utils/dom.js]: Dope implementation! Approving, but be careful of where this is utilized. If we fire it inside a 3rd party iframe (like IMA), it'll throw a cross origin iframe script blocking error.

File: src/js/utils/dom.js (hunk 1 above)
Comment (robwalch) [reply src/js/utils/dom.js]: I will add tests to [dom-tests.js](https://github.com/jwplayer/jwplayer/blob/master/test/unit/dom-test.js#L84)

Comment (robwalch) [reply src/js/utils/dom.js]: done

File: src/js/utils/dom.js [hunk 3]
```diff
@@ -49,17 +49,17 @@ export function htmlToParentElement(html) {
...
     }
     return element;
 }
 
 export function sanitizeElementAttributes(element) {
     const attributes = element.attributes;
-    for (let i = 0; i < attributes.length; i++) {
+    for (let i = attributes.length; i--;) {
```
Comment (robwalch) [inline src/js/utils/dom.js]: This is an important fix. `attributes.length` is decremented as attributes are removed, so we have to start at the end and cycle through in reverse to remove all matches.

**Pull request merged**